### PR TITLE
Adding 3 new .F90 tests per issue #81 and fixing related .c files - pt 4

### PR DESCRIPTION
**Pull request by tmh97**
All tests pass on CCE 12 on tulip (ftn)

- 3 new .f90 versions of previously existing .c tests
- adjustments to one .c file

### CODE REVIEW

File: tests/4.5/target_parallel/test_target_parallel.F90 [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+   CONTAINS
+      INTEGER FUNCTION target_parallel()
+         INTEGER :: errors, thread_id, i
+         INTEGER, DIMENSION(OMPVV_NUM_THREADS_DEVICE) :: num_threads
+         CHARACTER(len=400) :: threadMsg
+         errors = 0
+         thread_id = 0
+
+         !$omp target parallel num_threads(OMPVV_NUM_THREADS_DEVICE) map(from: num_threads(0:OMPVV_NUM_THREADS_DEVICE))
+            thread_id = omp_get_thread_num()
```
Comment (spophale) [inline tests/4.5/target_parallel/test_target_parallel.F90:35]: race on thread_id . better of line 36 is modified to:
num_threads(omp_get_thread_num()) = omp_get_num_threads()

File: tests/4.5/target_update/test_target_update_from.F90 [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+         c(i) = 0
+      END DO
+
+      !$omp target data map(to: a, b)
+         !$omp target
+            DO i = 1, N
+               b(i) = (a(i) + b(i))
+            END DO
+         !$omp end target
+         !$omp target update from(b)
```
Comment (spophale) [inline tests/4.5/target_update/test_target_update_from.F90:48]: why is it inside target data ?  
Better if it is stand alone (after end target data) IMO.

**Review by spophale — COMMENTED** [review]
Minor changes required.

File: tests/4.5/target_parallel/test_target_parallel.F90 (hunk 1 above)
Comment (tmh97) [reply tests/4.5/target_parallel/test_target_parallel.F90:35]: *Check .c file as well for same issue

File: tests/4.5/target_update/test_target_update_from.F90 (hunk 2 above)
Comment (tmh97) [reply tests/4.5/target_update/test_target_update_from.F90:48]: move update from outside!

**Review by spophale — APPROVED** [review]

**Review by nolanbaker31 — APPROVED** [review]
lgtm

**Comment by tmh97** [comment]
Closing as this is ready to go! Addresses issue #81

**Pull request merged**